MekHQ/src/mekhq/campaign/stratcon/StratconScenarioFactory.java

```diff
@@ -0,0 +1,189 @@
+                    if (!dynamicScenarioUnitTypeMap.containsKey(playerForceUnitType)) {
+                        dynamicScenarioUnitTypeMap.put(playerForceUnitType, new ArrayList<>());
+                    }
+                    
+                    dynamicScenarioUnitTypeMap.get(playerForceUnitType).add(template);
+                    
+                    dynamicScenarioNameMap.put(fileName, template);
+                }
+            } catch (Exception e) {
+                MekHQ.getLogger().error(String.format("Error loading file: %s", filePath), e);
+            }
+        }
+    }
+    
+    /**
+     * Retrieves a random scenario template in the appropriate location.
+     * @param location The location (ground/low atmo/space) category of the scenario.
+     * @return Random scenario template.
+     */
+    public static ScenarioTemplate getRandomScenario(MapLocation location) {
+    	return Utilities.getRandomItem(dynamicScenarioLocationMap.get(location)).clone();
+    }
+    
+    /**
+     * Retrieves a specific scenario given the key (file name)
+     */
+    public static ScenarioTemplate getSpecificScenario(String name) {
+        return dynamicScenarioNameMap.get(name).clone();
+    }
+    
+    /**
+     * Retrieves a random scenario template appropriate for the given unit type.
+     * This includes the more general ATB_MIX and ATB_AERO_MIX where appropriate
+     * @param unitType The desired unit type, as per megamek.common.UnitType
+     * @return Random scenario template.
+     */
+    public static ScenarioTemplate getRandomScenario(int unitType) {
+        int generalUnitType = convertSpecificUnitTypeToGeneral(unitType);
+        
+        // if the specific unit type doesn't have any scenario templates for it
+        // then we can't generate a scenario.
+        if (!dynamicScenarioUnitTypeMap.containsKey(unitType) &&
+            !dynamicScenarioUnitTypeMap.containsKey(generalUnitType)) {
+                MekHQ.getLogger().warning(String.format("No scenarios configured for unit type %d", unitType));
+                return null;
+        }
+        
+        List<ScenarioTemplate> jointList = new ArrayList<>();
+        
+        if (dynamicScenarioUnitTypeMap.containsKey(unitType)) {
+            jointList.addAll(dynamicScenarioUnitTypeMap.get(unitType));
+        }
+        
+        if (dynamicScenarioUnitTypeMap.containsKey(generalUnitType)) {
+            jointList.addAll(dynamicScenarioUnitTypeMap.get(generalUnitType));
+        }
+        
+        return Utilities.getRandomItem(jointList).clone();
+    }
+    
+    /**
+     * Get an allied or hostile facility scenario, depending on passed on parameter.
+     */
+    public static ScenarioTemplate getFacilityScenario(boolean allied) {
+        if (allied) {
+            return getSpecificScenario(ALLIED_FACILITY_SCENARIO);
+        } else {
+            return getSpecificScenario(HOSTILE_FACILITY_SCENARIO);
+        }
```

Comment:
```suggestion
        return getSpecificScenario(allied ? ALLIED_FACILITY_SCENARIO : HOSTILE_FACILITY_SCENARIO);
```

Reply:
Future expansion may feature a situation where a ternary operator is insufficient.